### ISSUE TITLE
32X Support?

### ISSUE THREAD

**Issue by eGax** [ekeeke/Genesis-Plus-GX#64]
Enhancement request for 32x support. Is this a possibility or totally out of the question?

**Comment by ekeeke** [comment]
It is possible but needs someone to do it. I personally do not have the motivation to devote my free time doing it but code submission are always welcomed.

As for 32x emulation on Wii, it has been discussed many times before but it would require rewriting a lot of the existing emulation code in Genesis Plus GX to at least hope reaching playable speed by making it optimized specifically for Wii hardware (thus making it not portable anymore), so it's not worth it.

**Comment by Kokokool** [comment]
5 years have passed since. Is there a chance now to have 32x support? I would definitely donate if theres progress in this issue

🙏🏻🙏🏻

**Comment by richterdeveloper** [comment]
@Kokokool Yeah!! I support the request too!! This emulator is great and awesome in high accuracy and compatibility.

**Comment by ner00** [comment]
> 5 years have passed since. Is there a chance now to have 32x support? I would definitely donate if theres progress in this issue
https://www.bountysource.com/issues/27901972-32x-support

**Comment by 60fpshacksrock** [comment]
https://github.com/ekeeke/Genesis-Plus-GX/issues/66#issuecomment-826193318

**Comment by SteveCraft27** [comment]
7 Years Of Waiting 32X Support On Genesis Plus GX. It's Better To Request SC-3000, SC-7000, Genesis, MegaCD, 32X, 32X CD On Gearsystem GitHub. Using Gearsystem 3.4.3 Update Request. Gearsystem Is SG-1000, Master System, Game Gear Emulator.

**Comment by SteveCraft27** [comment]
https://github.com/drhelius/Gearsystem/issues/65

**Comment by CraftFan23** [comment]
ekeeke doesn't have plan to emulate SC-3000 (Sega Computer 3000), SF-7000 (Sega Super Control Station 7000), Sega 32X.

**Comment by Gorilli09** [comment]
Can you make it for Windows only?

**Comment by Gorilli09** [comment]
> It is possible but needs someone to do it. I personally do not have the motivation to devote my free time on this but code submission are always welcomed.
> 
> As for 32x emulation on Wii, it has been discussed many times before but it would require rewriting a lot of the existing emulation code in Genesis Plus GX to at least hope reaching playable speed by making the code optimized specifically for Wii hardware (thus making it not portable anymore), so it's not really worth it.
> 

Can you make it for Windows only?

**Comment by Vamp898** [comment]
This maybe sounds like a silly question, but i assume the 32X Support from PicoDrive could not be ported, right?

**Comment by ner00** [comment]
> This maybe sounds like a silly question, but i assume the 32X Support from PicoDrive could not be ported, right?

Could it be ported? Possibly. I don't represent this project, but it seems that the developer was clear 10 years ago. 

Asking/suggesting to port 32X support from another emulator doesn't seem to make it easier, might even make it harder in a lot of ways. Truth is this issue was not closed since 10 years ago, maybe as a way to entice future collaboration. Yet nobody since then has committed any code in an attempt to integrate the support we would like this emulator to have for 32x.

Personally, I would be thrilled to have a community member finally submitting code to this project that supports 32x. Unfortunately everyone just drop request after request with no expectation to do any of the work. 

If I had the expertise I would do it, unfortunately I don't and that's why I am not asking for something that the dev explicitly requested but so far no one took upon this challenge.

**Comment by AlfredoDore** [comment]
> I have knowledge in C/C++ but zero knowledge in emulator development.
> 
> I am looking into how to start and found some good documentation about CHIP-8 but even when I manage to build a working CHIP-8, I assume it's still a long road to 32X (SH-2)
> 
> Porting an existing one (or using it as an base) might be more realistic... I guess. But if it would be that simple, I assume it would have been finished long ago.
> 
> I checked the source code of Picodrive and, obviously, it uses a lot of OpenSource projects code to begin with. The whole Z80 and Motorola 68000 Code originated from other projects.
> 
> The SH-2, they seem to have done that themselves and it's about 11000 lines of code so I assume that needs more knowledge then having an CHIP-8 done. Even when I start learning now, I won't be able to contribute any real code anytime soon :(

The modified RetroArch for Wii (whether RA or SS Haxeco) now includes Picodrive; I think you could look into that first.
